api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto

```diff
@@ -226,6 +241,19 @@ message ExternalProcessor {
   //    This work is currently tracked under https://github.com/envoyproxy/envoy/issues/33319.
   //
   bool observability_mode = 17;
+
+  oneof route_cache {
+    // Prevents clearing the route-cache when the
+    // :ref:`clear_route_cache <envoy_v3_api_field_service.ext_proc.v3.CommonResponse.clear_route_cache>`
+    // field is set in an external processor response.
+    bool disable_clear_route_cache = 11;
```

Comment:
We can not wrap existing field with oneof as it is a breaking API change. See [API guide](https://github.com/envoyproxy/envoy/blob/main/api/API_VERSIONING.md#backwards-compatibility).

We can use udpa.annotations.field_migrate).oneof_promotion here

Reply:
Another thing for consideration: do we really need oneof here?

Alternatively, we can mark `disable_clear_route_cache` as `[deprecated = true, (envoy.annotations.deprecated_at_minor_version) = "3.0"];` and add a note that `disable_clear_route_cache` is deprecated by RouteCacheAction RETAIN. 

You still need to handle `disable_clear_route_cache` as normal because deprecation will only happen when API is above 3.0 (which probably won't happen)

Reply:
Good catch!

I am thinking to do both, i.e, adding udpa.annotations.field_migrate).oneof_promotion, at same time put disable_clear_route_cache as deprecated.  This explicitly tells only one of these two should be configured, and RouteCacheAction is the one to use in the future.

Reply:
Technically, if `disable_clear_route_cache` is marked as deprecated, there will be no oneof in the future (as that deprecated field will be removed).  Thus, I think we can just choose either ONEOF or DEPRECATED here. 

One important thing please note, we should implement it in "oneof-style" as none of these annotation enforcements are effective currently.

Reply:
Let's keep oneof_promotion as its more informative.